pandas/core/generic.py

```diff
@@ -340,7 +340,7 @@ def attrs(self) -> dict[Hashable, Any]:
         return self._attrs
 
     @attrs.setter
-    def attrs(self, value: Mapping[Hashable, Any]) -> None:
+    def attrs(self, value: dict[Hashable, Any]) -> None:
         self._attrs = dict(value)
```

Comment:
shouldn't you leave the input annotation and then this is equiv of a cast? we do this in lots of places. why is this flagged?

Reply:
This is not a typing issue, If I understand it correctly, pyright flags this because the property is meant to act like a variable, which means that setting and getting the value of the variable should have the same type. 

Pyright doesn't like the following (independent of the function body):
```py
@property
def attrs(self) -> dict[Hashable, Any]:
    ...

@attrs.setter
def attrs(self, value: Mapping[Hashable, Any]) -> None:
    ...
```
Both should be a `Mapping` or both should be a `dict`.

I can see why pyright wants to encourage using property like a variable, but I'm also happy to close this PR if this is too controversial/not intended.